src/include/cursor_inline.h

```diff
@@ -52,7 +52,12 @@ __cursor_set_recno(WT_CURSOR_BTREE *cbt, uint64_t v)
 static WT_INLINE int
 __cursor_copy_release(WT_CURSOR *cursor)
 {
-    if (FLD_ISSET(S2C(CUR2S(cursor))->debug_flags, WT_CONN_DEBUG_CURSOR_COPY)) {
+    /*
+     * Make a redundant test first, since it should always fail for high performance situations and
+     * we can exit the function quickly.
+     */
+    if (F_ISSET(cursor, WT_CURSTD_DEBUG_COPY_KEY | WT_CURSTD_DEBUG_COPY_VALUE) &&
+      FLD_ISSET(S2C(CUR2S(cursor))->debug_flags, WT_CONN_DEBUG_CURSOR_COPY)) {
```

Comment:
It turns out that in addition to the `(F_ISSET(cursor, WT_CURSTD_DEBUG_COPY_KEY | WT_CURSTD_DEBUG_COPY_VALUE)` test, we also need to check  `FLD_ISSET(S2C(CUR2S(cursor))->debug_flags, WT_CONN_DEBUG_CURSOR_COPY)` like the original code.  Or else wt2323_join fails under ASAN.

I think that DEBUG_COPY_KEY or DEBUG_COPY_VALUE can get set even if WT_CONN_DEBUG_CURSOR_COPY is not set.   That happens at the end of __wt_cursor_set_valuev/keyv.